source/common/json/json_loader.cc

```diff
@@ -1,178 +1,194 @@
 #include "json_loader.h"
 
-// Do not let jansson leak outside of this file.
+// Do not let RapidJson leak outside of this file.
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
-#include "jansson.h"
+#include "rapidjson/document.h"
+#include "rapidjson/error/en.h"
+#include "rapidjson/istreamwrapper.h"
 #pragma GCC diagnostic pop
 
 namespace Json {
 
-FileLoader::FileLoader(const std::string& file_path) {
-  json_error_t error;
-  json_ = json_load_file(file_path.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {} line: {} column: {}", error.text,
-                                error.line, error.column));
+/**
+ * Implementation of AbstractObject.
+ */
+class AbstractObjectImpl : public AbstractObject {
+public:
+  AbstractObjectImpl(const rapidjson::Document& document, const std::string& name) : name_(name) {
+    document_.CopyFrom(document, document_.GetAllocator());
   }
-}
-
-FileLoader::~FileLoader() { json_decref(json_); }
 
-StringLoader::StringLoader(const std::string& json) {
-  json_error_t error;
-  json_ = json_loads(json.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {}", error.text));
+  AbstractObjectImpl(const rapidjson::Value& value, const std::string& name) : name_(name) {
+    document_.CopyFrom(value, document_.GetAllocator());
   }
-}
 
-StringLoader::~StringLoader() { json_decref(json_); }
+  // Empty Object
+  AbstractObjectImpl(const std::string& name) : name_(name) {}
 
-Object::EmptyObject Object::empty_;
-
-Object::EmptyObject::EmptyObject() : json_(json_object()) {}
+  std::vector<AbstractObjectPtr> asObjectArray() const override {
+    if (!document_.IsArray()) {
+      throw Exception(fmt::format("'{}' is not an array", name_));
+    }
 
-Object::EmptyObject::~EmptyObject() { json_decref(json_); }
+    std::vector<AbstractObjectPtr> object_array;
+    for (rapidjson::SizeType i = 0; i < document_.Size(); i++) {
+      object_array.emplace_back(new AbstractObjectImpl(document_[i], name_ + " (array item)"));
+    }
 
-std::vector<Object> Object::asObjectArray() const {
-  if (!json_is_array(json_)) {
-    throw Exception(fmt::format("'{}' is not an array", name_));
+    return object_array;
   }
 
-  std::vector<Object> object_array;
-  for (size_t i = 0; i < json_array_size(json_); i++) {
-    object_array.emplace_back(json_array_get(json_, i), name_ + " (array item)");
+  bool getBoolean(const std::string& name) const override {
+    if (!document_[name.c_str()].IsBool()) {
```

Comment:
not to familiar with api, but can we
1) auto &blah = document_[name.c_str()];
and query internal map or something once.

if (!blah.IsBool()) {
}

return blah.GetBool();